vault/identity_store_oidc.go

```diff
@@ -0,0 +1,743 @@
+		// },
+
+		// {
+		{
+			Pattern: "oidc/key/" + framework.GenericNameRegex("name"),
+			Fields: map[string]*framework.FieldSchema{
+				"name": &framework.FieldSchema{
+					Type:        framework.TypeString,
+					Description: "Name of the key",
+				},
+
+				"rotation_period": &framework.FieldSchema{
+					Type:        framework.TypeString,
+					Description: "How often to generate a new keypair. Defaults to 6h",
+				},
+
+				"verification_ttl": &framework.FieldSchema{
+					Type:        framework.TypeString,
+					Description: "Controls how long the public portion of a key will be available for verification after being rotated. Defaults to the current rotation_period, which will provide for a current key and previous key.",
+				},
+
+				"algorithm": &framework.FieldSchema{
+					Type:        framework.TypeString,
+					Description: "Signing algorithm to use. This will default to RS256, and is currently the only allowed value.",
+				},
+			},
+			Callbacks: map[logical.Operation]framework.OperationFunc{
+				logical.UpdateOperation: i.handleOIDCCreateKey(),
+				logical.ReadOperation:   i.handleOIDCReadKey(),
+				logical.DeleteOperation: i.handleOIDCDeleteKey(),
+			},
+			HelpSynopsis:    "oidc/key/:key help synopsis here",
+			HelpDescription: "oidc/key/:key help description here",
+		},
+		{
+			Pattern: "oidc/key/" + framework.GenericNameRegex("name") + "/rotate/?$",
+			Fields: map[string]*framework.FieldSchema{
+				"name": &framework.FieldSchema{
+					Type:        framework.TypeString,
+					Description: "Name of the key",
+				},
+				"verification_ttl": &framework.FieldSchema{
+					Type:        framework.TypeString,
+					Description: "Controls how long the public portion of a key will be available for verification after being rotated. Setting verification_ttl here will override the verification_ttl set on the key.",
+				},
+			},
+			Callbacks: map[logical.Operation]framework.OperationFunc{
+				logical.UpdateOperation: i.handleOIDCRotateKey(),
+			},
+			HelpSynopsis:    "oidc/key/:key/rotate help synopsis here",
+			HelpDescription: "oidc/key/:key/rotate help description here",
+		},
+		{
+			Pattern: "oidc/key/?$",
+			Callbacks: map[logical.Operation]framework.OperationFunc{
+				logical.ListOperation: i.handleOIDCListKeys(),
+			},
+			HelpSynopsis:    "list oidc/key/ help synopsis here",
+			HelpDescription: "list oidc/key/ help description here",
+		},
+		{
+			Pattern: "oidc/well-known/certs/?$",
+			Callbacks: map[logical.Operation]framework.OperationFunc{
+				logical.ReadOperation: i.handleOIDCReadCerts(),
+			},
+			HelpSynopsis:    "read oidc/well-known/certs/ help synopsis here",
+			HelpDescription: "read oidc/well-known/certs/ help description here",
+		},
+	}
+}
+
+// handleOIDCCreateKey is used to create a new key
+func (i *IdentityStore) handleOIDCCreateKey() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
```

Comment:
Any reason for the anonymous func for these handler? Why not just have `handleOIDCCreateKey` be the handler?

Reply:
nope - was looking at a template that did this.